pytests/src/datetime.rs

```diff
@@ -37,6 +37,7 @@ fn make_time<'p>(
 }
 
 #[pyfunction]
+#[pyo3(signature = (hour, minute, second, microsecond, tzinfo, fold))]
```

Comment:
Why is this annotation now necessary?

Reply:
That is because `tzinfo` is an `Option` and `fold` is not an option; the error message without the signature is this:

```
warning: use of deprecated constant `pyo3::impl_::deprecations::REQUIRED_ARGUMENT_AFTER_OPTION`: required arguments after an `Option<_>` argument are ambiguous and being phased out
         = help: add a `#[pyo3(signature)]` annotation on this function to unambiguously specify the default values for all optional parameters
  --> pytests/src/datetime.rs:48:5
   |
48 |     fold: bool,
   |     ^^^^
   |
   = note: `#[warn(deprecated)]` on by default
```

By adding the signature which doesn't give `tzinfo` a default, it makes it unambiguous that `tzinfo` is a required argument in this context.

To be honest, I should probably split this change out into a separate PR.

Reply:
Pushed #2703 to review that change later.